lib/policial/client.rb

```diff
@@ -0,0 +1,41 @@
+module Policial
+  # Public: Starting with an Octokit client and a pull request,
+  # it checks all changes introduced looking for style guide violations. It
+  # also accuses all present violations through comments on all relevant lines
+  # in the pull request.
+  class Client
+    attr_accessor :violations
+    attr_reader :octokit, :pull_request
+
+    def initialize(octokit = nil)
+      @octokit = octokit || Octokit
+    end
+
+    def investigate
+      @violations ||= StyleChecker.new(pull_request).violations
+    end
+
+    def accuse
+      return if violations.nil?
+
+      commenter = Commenter.new(pull_request)
+
+      violations.each do |violation|
+        if accusation_policy.allowed_for?(violation)
+          commenter.comment_violation(violation)
+        end
+      end
+    end
+
+    def pull_request=(pull_request)
+      @pull_request = pull_request
+      @pull_request.octokit = @octokit
```

Comment:
It feels weird to have this side effect of assigning the pull request. I was thinking if we could not change this method to a factory method that will create the `PullRequest` object. Something like:

```ruby
def file(content)
  @pull_request = PullRequest.new(content.merge(octokit: ocktokit)
```

And the usage flow would be simpler, and users would just have to know about the `Detective` class and `PullRequest` could be changed to be internal API.

```ruby
policial_client = Policial::Client.new(octokit)
policial_client.file(repo: 'volmer/my_repo', number: 3, head_sha: 'headsha')
```